src/oss-diff/DiffTool.cs

```diff
@@ -0,0 +1,128 @@
+        }
+
+        public async Task RunAsync(Options options)
+        {
+            if (options.Targets.Count() != 2)
+            {
+                Logger.Error("Must provide exactly two packages to diff.");
+                return;
+            }
+            (PackageURL purl1, PackageURL purl2) = (new PackageURL(options.Targets.First()), new PackageURL(options.Targets.Last()));
+            var manager = ProjectManagerFactory.CreateProjectManager(purl1,options.DownloadDirectory);
+            var locations = await manager.DownloadVersion(purl1, true, options.UseCache);
+
+            var manager2 = ProjectManagerFactory.CreateProjectManager(purl2, options.DownloadDirectory);
+            var locations2 = await manager2.DownloadVersion(purl2, true, options.UseCache);
+
+            Dictionary<string, (string, string)> files = new Dictionary<string, (string, string)>();
+            foreach(var directory in locations)
+            {
+                foreach (var file in Directory.EnumerateFiles(directory, "*", SearchOption.AllDirectories))
+                {
+                    var contents = File.ReadAllText(file);
+                    files.Add(file.Substring(directory.Length), (contents, string.Empty));
+                }
+            }
+
+            foreach (var directory in locations2)
+            {
+                foreach (var file in Directory.EnumerateFiles(directory, "*", SearchOption.AllDirectories))
+                {
+                    var contents = File.ReadAllText(file);
+                    var key = file.Substring(directory.Length);
+
+                    if (files.ContainsKey(key))
+                    {
+                        var existing = files[key];
+                        existing.Item2 = contents;
+                        files[key] = existing;
+                    }
+                    else
+                    {
+                        files[key] = (string.Empty, contents);
+                    }
+                }
+            }
+
+            foreach (var filePair in files)
+            {
+                var diff = InlineDiffBuilder.Diff(filePair.Value.Item1, filePair.Value.Item2);
+                Console.WriteLine(filePair.Key);
+                var savedColor = Console.ForegroundColor;
+                foreach (var line in diff.Lines)
+                {
+                    switch (line.Type)
+                    {
+                        case ChangeType.Inserted:
+                            Console.ForegroundColor = ConsoleColor.Green;
+                            Console.Write("+ ");
+                            break;
+                        case ChangeType.Deleted:
+                            Console.ForegroundColor = ConsoleColor.Red;
+                            Console.Write("- ");
+                            break;
+                        default:
+                            Console.ForegroundColor = ConsoleColor.Gray; // compromise for dark or light background
```

Comment:
You'll probably want to limit the context to N  (default 3?) lines before/after a change so the output isn't huge. 

Also, not sure if .NET handles this automatically, but if the environment variable NO_COLOR is set, just output plain text.

Reply:
I'll try switching to one of the libraries listed here like Pastel that auto support NO_COLOR

https://no-color.org